dev-python/sentinels/sentinels-1.0.0.ebuild

```diff
@@ -0,0 +1,32 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,8} )
+DISTUTILS_IN_SOURCE_BUILD=1
+
+inherit distutils-r1 eutils
+
+DESCRIPTION="Python Sentinels class which are objects with special meanings."
+HOMEPAGE="https://pypi.org/project/sentinels"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+BDEPEND="
+"
+
+DEPEND="
+	test? (
+		dev-python/nose[${PYTHON_USEDEP}]
+	)"
+
+python_test() {
+	distutils_install_for_testing
+	nosetests -v -w ./tests || die "Testsuite failed under ${EPYTHON}"
+}
```

Comment:
we have new python eclass functions
instead of defining RESTRICT, DEPEND and python_test, you can add a single line
`distutils_enable_tests nose`

https://dev.gentoo.org/~mgorny/python-guide/distutils.html#enabling-tests

hence the ebuild shortens to
```bash
# Copyright 1999-2020 Gentoo Authors
# Distributed under the terms of the GNU General Public License v2

EAPI=7

PYTHON_COMPAT=( python3_{6,8} )

inherit distutils-r1 eutils

DESCRIPTION="Python Sentinels class which are objects with special meanings."
HOMEPAGE="https://pypi.org/project/sentinels"
SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"

LICENSE="BSD"
SLOT="0"
KEYWORDS="~amd64"

distutils_enable_tests nose
```

Reply:
it also puts the test output in one place :)
```
>>> Test phase: dev-python/sentinels-1.0.0                                                      
 * python3_6: running distutils-r1_run_phase python_test                                                                                                                                                            
test__readme_doctests (test__readme_doctest.ReadMeDocTest) ... ok                                     
                                                                                                                                                                                                                    
----------------------------------------------------------------------                 
Ran 1 test in 0.008s                                                                
                                                                                                                                                                                                                    
OK                                                                 
 * python3_7: running distutils-r1_run_phase python_test               
test__readme_doctests (test__readme_doctest.ReadMeDocTest) ... ok
                                                                                                          
----------------------------------------------------------------------                
Ran 1 test in 0.010s                                                                      
                                                                                                          
OK                                                                                                      
 * python3_8: running distutils-r1_run_phase python_test                                                                                                                                                            
test__readme_doctests (test__readme_doctest.ReadMeDocTest) ... ok                                                                                                                                                   
                                                                                                                                                                                                                                                                                                                                                                                                                                         
----------------------------------------------------------------------                                                                                                                                                                                                                                                                                                                                                                   
Ran 1 test in 0.010s                                                                                                                                                                                                
                                                                                                                                                                                                                    
OK 
```